splunk_connect_for_snmp_traps/manager/trap_server.py

```diff
@@ -19,73 +27,101 @@ def __init__(self, server_config):
-            response = requests.post(url=endpoint, json=data, headers=headers, verify=self._hec_config.is_ssl_enabled())
-            logger.debug(f'Response code is {response.status_code}')
+            headers = {
+                "Authorization": f"Splunk {self._hec_config.get_authentication_token()}"
+            }
+            data = {"sourcetype": "trap-server", "event": variables_binds}
+            logger.debug(f"Posting trap to HEC using {endpoint} and headers {headers}")
+            response = requests.post(
+                url=endpoint,
+                json=data,
+                headers=headers,
+                verify=self._hec_config.is_ssl_enabled(),
+            )
+            logger.debug(f"Response code is {response.status_code}")
 
     # Register a callback to be invoked at specified execution point of
     # SNMP Engine and passed local variables at code point's local scope
     # noinspection PyUnusedLocal,PyUnusedLocal
     def request_observer(self, snmp_engine, execution_point, variables, callback_ctx):
         logger.debug(f'Raw data is "{variables}"')
-        logger.debug('Execution point: %s' % execution_point)
+        logger.debug("Execution point: %s" % execution_point)
         logger.debug(
-            '* transportDomain: %s'
-            % '.'.join([str(x) for x in variables['transportDomain']])
+            "* transportDomain: %s"
+            % ".".join([str(x) for x in variables["transportDomain"]])
         )
         logger.debug(
-            '* transportAddress: %s'
-            % '@'.join([str(x) for x in variables['transportAddress']])
+            "* transportAddress: %s"
+            % "@".join([str(x) for x in variables["transportAddress"]])
         )
-        logger.debug('* securityModel: %s' % variables['securityModel'])
-        logger.debug('* securityName: %s' % variables['securityName'])
-        logger.debug('* securityLevel: %s' % variables['securityLevel'])
-        logger.debug('* contextEngineId: %s' % variables['contextEngineId'].prettyPrint())
-        logger.debug('* contextName: %s' % variables['contextName'].prettyPrint())
-        logger.debug('* PDU: %s' % variables['pdu'].prettyPrint())
+        logger.debug("* securityModel: %s" % variables["securityModel"])
+        logger.debug("* securityName: %s" % variables["securityName"])
+        logger.debug("* securityLevel: %s" % variables["securityLevel"])
+        logger.debug(
+            "* contextEngineId: %s" % variables["contextEngineId"].prettyPrint()
+        )
+        logger.debug("* contextName: %s" % variables["contextName"].prettyPrint())
+        logger.debug("* PDU: %s" % variables["pdu"].prettyPrint())
 
     # Callback function for receiving notifications
     # noinspection PyUnusedLocal,PyUnusedLocal,PyUnusedLocal
-    def snmp_callback_function(self, snmp_engine, state_reference, context_engine_id, context_name, var_binds,
-                               callback_ctx):
+    def snmp_callback_function(
```

Comment:
Sometimes I like the old formatting. I like double quotes but this multiline function definition looks far from clean sometimes.

Reply:
Affirmative.